### PR TITLE
Abstract tests single function

### PR DESCRIPTION
**Pull request by ralphtheninja**

### CODE REVIEW

File: test/close-test.js [hunk 1]
```diff
@@ -1,9 +0,0 @@
-const test = require('tape')
-const testCommon = require('./common')
-const abstract = require('abstract-leveldown/test/close-test')
-
-test('setUp', testCommon.setUp)
```
Comment (ralphtheninja) [inline test/close-test.js]: Note that we're skipping `setUp` and `tearDown` now.

Comment (vweevers) [inline test/close-test.js]: Hmm, we should add a setup+teardown to https://github.com/Level/abstract-leveldown/blob/master/test/close-test.js

Comment (ralphtheninja) [reply test/close-test.js]: And an `.all()` method.

Comment (ralphtheninja) [reply test/close-test.js]: I'll make an issue and hack it up.

File: test/common.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
+const test = require('tape')
 const tempy = require('tempy')
 const leveldown = require('..')
 
-testCommon.factory = function () {
-  return leveldown(tempy.directory())
-}
-
-module.exports = testCommon
+module.exports = require('abstract-leveldown/test/common')({
```
Comment (vweevers) [inline test/common.js]: Ah this comes in handy

**Comment by ralphtheninja** [comment]
I'll update this PR once https://github.com/Level/abstract-leveldown/issues/278 has been merged

**Comment by ralphtheninja** [comment]
I get the following error now:

screenshot from 2018-07-13 18-09-28

**Comment by vweevers** [comment]
@ralphtheninja we have to update chained batch.

**Comment by vweevers** [comment]
Hmm I now see that `levelup` doesn't take options in `Batch.write` either: https://github.com/Level/levelup/blob/c3845eea4547a829b05548c092b84d5cc30b8e46/lib/batch.js#L50

So perhaps we should just remove the `options` argument from `abstract-leveldown`?

**Comment by ralphtheninja** [comment]
> So perhaps we should just remove the options argument from abstract-leveldown?

I think it's fine to pass on options to `_write()`. We can either omit passing it to the C++ layer (just ignoring options) in `leveldown` or change on the C++ side.

**Comment by vweevers** [comment]
It'd be dead code though (effectively), unless we also add an options argument to levelup.

**Comment by ralphtheninja** [comment]
> It'd be dead code though (effectively), unless we also add an options argument to levelup.

So lets do that :)

**Comment by ralphtheninja** [comment]
> It'd be dead code though (effectively), unless we also add an options argument to levelup.

It could default to `{}` in `levelup` so we can release it as a patch.

**Comment by vweevers** [comment]
Alright, SGTM

**Comment by ralphtheninja** [comment]
@vweevers Make an issue?

**Comment by vweevers** [comment]
=> https://github.com/Level/levelup/issues/603

**Pull request merged**